eng/automation/generate.py

```diff
@@ -199,7 +201,7 @@ def sdk_automation_autorest(config: dict) -> List[dict]:
                     ],
                     "readmeMd": [readme],
                     "artifacts": ["{0}/pom.xml".format(output_folder)]
-                    + [jar for jar in glob.glob("{0}/target/*.jar".format(output_folder))],
+                    + [jar for jar in glob.glob("{0}/target/*.jar".format(output_folder))] if succeeded else [],
```

Comment:
Had to set to `[]` if failed, otherwise sdkautomation would fail and override the original error:
![image](https://github.com/user-attachments/assets/20c4b77e-d095-409d-b529-cb8b21fdaf1e)